app/controllers/v1/matches_controller.rb

```diff
@@ -0,0 +1,14 @@
+class V1::MatchesController < ApplicationController
+
+  # /matches?competition_id=:id&user_id=:id
+  def index
+    @user = User.find_by(id: params[:user_id]) || current_user
+    @competition = Competition.find_by(id: params[:competition_id])
+    @matches =
+      if @competition
+        policy_scope(Match).where(group: @competition.groups).order(:kickoff_time)
+      else
+        policy_scope([:user, Match]).order(:kickoff_time)
+      end
```

Comment:
so yeah i was staring at this quite a bit last night. It's quite confusing because matches done actually belong to a user. only the predictions. so in the jbuilder view, i was loading the prediction for each user with each match (the long way). Cause they should still see matches even if they don't have a competition. 

the scoped user way (without a competition) would only show matches where a user has a prediction

Reply:
Right I see what you mean... 🤔 

Do we ever want to only show a user's predictions? The way I see it, we'd always show all the matches (for a competition or round for example) and include the prediction for the matches that happen to have one? In this case, `user_id` would not technically be a filter, but an optional argument to fetch matching predictions?

Reply:
a prediction alone doesnt make any sense. we definitely always want matches + predictions. 

> In this case, user_id would not technically be a filter, but an optional argument to fetch matching predictions (if present)?
I'm a little confused by this. The only time we'd pass a user into this is when we're looking at another person's predictions

Reply:
Yes indeed, my question is about this:

> the scoped user way (without a competition) would only show matches where a user has a prediction

Would we ever show only the matches where the user has predictions? Shouldn't we always show all of the matches (for a competition/round) and indicate whether or not the user is missing a prediction?

In other words, would we ever show a page where we hide the matches where the user is missing predictions? (I feel like these matches should still show up nevertheless, but with a "missing prediction" indicator?

Reply:
> I'm a little confused by this. The only time we'd pass a user into this is when we're looking at another person's predictions

You're right... Ok then yes, it would make indeed sense then that we default the `user_id` to the current user's 👍

Reply:
As agreed during our Slack call, we'll handle the multi-competitions case later